src/main/java/com/commercetools/sync/types/TypeSync.java

```diff
@@ -99,54 +98,35 @@ public TypeSync(@Nonnull final TypeSyncOptions typeSyncOptions) {
     @Override
     protected CompletionStage<TypeSyncStatistics> processBatch(@Nonnull final List<TypeDraft> batch) {
 
-        final Set<TypeDraft> validTypeDrafts = batch.stream().filter(this::validateDraft).collect(toSet());
+        final ImmutablePair<Set<TypeDraft>, Set<String>> result =
+            batchValidator.validateAndCollectReferencedKeys(batch);
 
-        if (validTypeDrafts.isEmpty()) {
+        final Set<TypeDraft> validDrafts = result.getLeft();
+        if (validDrafts.isEmpty()) {
             statistics.incrementProcessed(batch.size());
-            return completedFuture(statistics);
-        } else {
-            final Set<String> keys = validTypeDrafts.stream().map(TypeDraft::getKey).collect(toSet());
-
-            return typeService
-                .fetchMatchingTypesByKeys(keys)
-                .handle(ImmutablePair::new)
-                .thenCompose(fetchResponse -> {
-                    final Set<Type> fetchedTypes = fetchResponse.getKey();
-                    final Throwable exception = fetchResponse.getValue();
-
-                    if (exception != null) {
-                        final String errorMessage = format(CTP_TYPE_FETCH_FAILED, keys);
-                        handleError(errorMessage, exception, keys.size());
-                        return CompletableFuture.completedFuture(null);
-                    } else {
-                        return syncBatch(fetchedTypes, validTypeDrafts);
-                    }
-                })
-                .thenApply(ignored -> {
-                    statistics.incrementProcessed(batch.size());
-                    return statistics;
-                });
+            return CompletableFuture.completedFuture(statistics);
         }
-    }
+        final Set<String> validKeys = result.getRight();
 
-    /**
-     * Checks if a draft is valid for further processing. If so, then returns {@code true}. Otherwise handles an error
-     * and returns {@code false}. A valid draft is a {@link TypeDraft} object that is not {@code null} and its
-     * key is not empty.
-     *
-     * @param draft nullable draft
-     * @return boolean that indicate if given {@code draft} is valid for sync
-     */
-    private boolean validateDraft(@Nullable final TypeDraft draft) {
-        if (draft == null) {
-            handleError(TYPE_DRAFT_IS_NULL, null, 1);
-        } else if (isBlank(draft.getKey())) {
-            handleError(TYPE_DRAFT_HAS_NO_KEY, null, 1);
-        } else {
-            return true;
-        }
+        return typeService
+            .fetchMatchingTypesByKeys(validKeys)
```

Comment:
```suggestion
            .cacheKeysToIds(validKeys)
```

Reply:
No reference to cache on type sync, change only related to the moving validation logic to TypeBatchValidator.